zfs/zfs.go

```diff
@@ -1032,6 +1037,34 @@ type RecvOptions struct {
 	RollbackAndForceRecv bool
 	// Set -s flag used for resumable send & recv
 	SavePartialRecvState bool
+	InheritProperties    []Property
+	OverrideProperties   map[Property]string
+}
+
+type Property string
+
+func (p Property) Validate() error {
+	// see zfsprops(8), section "User Properties"
+	if len(p) < 1 {
+		return fmt.Errorf("property name can't be empty")
+	}
+	if len(p) > 256 {
+		return fmt.Errorf("property name longer than 256 characters")
+	}
+	if p[0] == '-' {
+		return fmt.Errorf("property name can't start with '-'")
```

Comment:
- [ ] `can't`
- [ ] this seems fairly arbitrary, please put a comment here that explains why we only need to check for `-`

Reply:
Literally from the above mentioned man page section. I've moved the comment above the function so it's clearer it applies to the whole function.